dissect/volume/vinum/vinum.py

```diff
@@ -0,0 +1,340 @@
+            if not disk.active:
+                # Assuming here that if a disk is marked as inactive, it's
+                # configuration is also old and/or possibly inaccurate.
+                continue
+
+            disks_by_name[disk.id] = disk
+
+            # For now we only use the first config block. The second config block
+            # could be useful/used as a fallback if for instance the first one
+            # fails to parse.
+            #
+            # The self.header.label.last_update timestamp is used to see if this
+            # disk contains the latest/newest config of the disks in a set. The
+            # disk is ignored if there is no header (which in our case would have
+            # raised a ValueError) or if the state is not set to GV_DRIVE_UP (an
+            # internal kernel state which we don't have and thus can ignore).
+            #
+            # Plexes with a non-existing Volume and SDs with a non-existing Plex or
+            # Drive (VinumPhysicalDisk) can not be used. They result in an error in
+            # the FreeBSD kernel code.
+            config_data = parse_vinum_config(disk.config_time, disk.config)
+
+            for config_type, new_items in config_data.items():
+                cur_config = config[config_type]
+
+                for new_item in new_items:
+                    if cur_item := cur_config.get(new_item.name):
+                        if new_item.timestamp > cur_item.timestamp:
+                            cur_config[new_item.name] = new_item
+                    else:
+                        cur_config[new_item.name] = new_item
+
+        # plexes_by_name contains all *active* plexes
+        plexes_by_name: ByName[Plex] = dict()
+
+        # plexes_by_volume contains all *active* plexes grouped by the name of
+        # the volume they belong to
+        plexes_by_volume: DefaultByName[list[Plex]] = defaultdict(list)
+        # sds_by_plex_by_volume contains all sds for all *active* plexes
+        # grouped by plex and volume name
+        sds_by_plex_by_volume: DefaultByName[DefaultByName[list[SD]]] = defaultdict(lambda: defaultdict(list))
+
+        # Check whether the found plexes are viable and filter them out if they
+        # are not.
+        # Note that if a complete Volume is down, we still try to use it (we don't check on Volume.state).
+        for plex in config["plexes"].values():
+            if config["volumes"].get(plex.volume):
+                # A plex in the DOWN state is probably not fit to use
+                if plex.state != PlexState.DOWN:
+                    if plex.org != PlexOrg.DISORG:
+                        plexes_by_name[plex.name] = plex
+                        plexes_by_volume[plex.volume].append(plex)
+                    else:
+                        log.warning("Plex %r has an unknown organistaion, ignoring plex", plex.name)
+                else:
+                    log.warning("Plex %r is down, ignoring plex", plex.name)
```

Comment:
What does it mean when a plex is "down"? In the case of Dissect, would it not still be interesting to "open" "down" plexes?

Reply:
I'm not sure how "broken" Plexes are that are down. If we do try to use them it could especially interfere with mirrored sets where one of the mirrors is still up/correct.